### PR TITLE
Temporary fix for checkbox input not set to false

### PR DESCRIPTION
**Pull request by srascar**
This PR fix issues on api when boolean values are never set to false

### CODE REVIEW

File: Form/EventListener/FixCheckboxDataListener.php [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+namespace Sonata\CoreBundle\Form\EventListener;
+
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\Form\Extension\Core\DataTransformer\BooleanToStringTransformer;
+use Symfony\Component\Form\FormEvent;
+use Symfony\Component\Form\FormEvents;
+
+/**
+ * Fix an issue where value was never set
+ * to false in specific cases.
```
Comment (rande) [inline Form/EventListener/FixCheckboxDataListener.php]: Please explain a bit more

**Comment by srascar** [comment]
@rande

File: Form/EventListener/FixCheckboxDataListener.php [hunk 2]
```diff
@@ -0,0 +1,62 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
```
Comment (amine26tz) [inline Form/EventListener/FixCheckboxDataListener.php]: wrong header

File: Form/EventListener/FixCheckboxDataListener.php [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+/**
+ * Using BooleanToStringTransform in a checkbox form type
+ * will set false value to '0' instead of null which will end up
+ * returning true value when the form is bind
+ *
+ * Class FixCheckboxDataListener
+ *
+ * @author  Sylvain Rascar <rascar.sylvain@gmail.com>
+ *
+ * @package Symfony\Component\Form\Extension\Core\EventListener
```
Comment (amine26tz) [inline Form/EventListener/FixCheckboxDataListener.php]: wrong package

File: Tests/Form/EventListener/FixCheckboxDataListenerTest.php [hunk 4]
```diff
@@ -0,0 +1,55 @@
+<?php
+/*
+ * This file is part of the Symfony package.
```
Comment (amine26tz) [inline Tests/Form/EventListener/FixCheckboxDataListenerTest.php]: wrong header

**Comment by rande** [comment]
Can you squash commit ?

**Comment by rande** [comment]
@srascar can you squash this PR ?

**Pull request merged**